src/framework/audio/iplayback.h

```diff
@@ -19,36 +19,35 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
-#ifndef MU_AUDIO_IAUDIOENGINE_H
-#define MU_AUDIO_IAUDIOENGINE_H
-
-#include <string>
+#ifndef MU_AUDIO_ISEQUENCER_H
+#define MU_AUDIO_ISEQUENCER_H
 
 #include "modularity/imoduleexport.h"
-
-#include "ret.h"
 #include "async/channel.h"
-#include "isequencer.h"
-#include "iaudiodriver.h"
-#include "internal/iaudiobuffer.h"
-#include "imixer.h"
-#include "isynthesizer.h"
+#include "async/promise.h"
+
+#include "iplayers.h"
+#include "itracks.h"
+#include "iaudioio.h"
+#include "audiotypes.h"
 
 namespace mu::audio {
-class IAudioEngine : MODULE_EXPORT_INTERFACE
+class IPlayback : MODULE_EXPORT_INTERFACE
 {
-    INTERFACE_ID(IAudioEngine)
+    INTERFACE_ID(IPlayback)
 
 public:
-    virtual ~IAudioEngine() = default;
+    virtual ~IPlayback() = default;
 
-    virtual bool isInited() const = 0;
-    virtual async::Channel<bool> initChanged() const = 0;
-    virtual unsigned int sampleRate() const = 0;
-    virtual ISequencerPtr sequencer() const = 0;
-    virtual IMixerPtr mixer() const = 0;
-    virtual IAudioBufferPtr buffer() const = 0;
-    virtual void setAudioBuffer(IAudioBufferPtr) = 0;
+    virtual async::Promise<TrackSequenceId> addSequence() = 0;
+    virtual async::Promise<TrackSequenceIdList> sequenceIdList() const = 0;
+    virtual void removeSequence(const TrackSequenceId id) = 0;
+
+    virtual ITracksPtr tracks() const = 0;
+    virtual IPlayersPtr players() const = 0;
+    virtual IAudioIOPtr audioIO() const = 0;
```

Comment:
I looked at how to use it, I believe this is not convenient and not clear. May be better to make one interface for all?

Reply:
It is difficult to understand what needs to be done to use this interface and in what order
Is it true that I can add a track with the given input and output parameters - for this I need to use the `ITracks::addTrack` interface?
And I can change the in and out parameters - for this I need to use the `IAudioIO::setInputParams`,  `IAudioIO::setOutputParams` interface?